docs/source/en/hf_quantizer.md

```diff
@@ -0,0 +1,67 @@
+# Build a new `HFQuantizer` class to add quantization support for a new quantization method.
+
+Through this document, you will learn how to work on a transformers integration of a new quantization method. Note that currently the `HFQuantizer` is not meant to be used for any PyTorch module, but you should rather see it as an internal utility class that is used in the core modeling code to easily quantize transformers models with different SoTA approaches (e.g. QLoRA, GPTQ, LLM.int8, AWQ, ...). 
+
+
+## Pre-requisities 
+
+Before you start integrating a new quantization method into transformers, make sure that the method you are trying to add meet the following pre-requisities. Note we only support quantization methods that can be run with PyTorch modules for now.
+
+- The quantization method is available through a Python package that is pip-installable by anyone (it is also fine if you can only install the package from source), ideally with pre-compiled kernels included in the pip package.
+- The method can at least run on a commonly-used hardware (CPU, GPU, ..).
+- The method is wrapped in a `nn.Module` (e.g. `Linear8bitLt`, `Linear4bit`). Ideally your quantized linear layer should have the following definition
+```py
+class Linear4bit(nn.Module):
+    def __init__(self, ...):
+        ...
+    
+    def forward(self, x):
+        return my_4bit_kernel(x, self.weight, self.bias)
+```
+That way, transformers models can be easily quantizable by simply replacing some instances of `nn.Linear` with a target class.
+- Ideally the quantization method should be serializable, i.e. you can save the quantized weights locally or push them on the Hub.
+- Make sure the package that contains the quantization kernels / primitive is mature enough (e.g. no frequent breaking changes).
+
+Note that for some quantization methods it is a strong requirement to "pre-quantize" the models through data calibration (e.g. AWQ). In that case, we prefer to support only inference through transformers and let third-party libraries maintained by the ML community deal with the model quantization itself.
+
+## How should I get started?
+
+0- 📕 Create a new quantization config class inside `src/transformers/utils/quantization_config.py`, and make sure to expose that new quantization config inside transformers main init, by adding it on the `_import_structure` object of `src/transformers/__init__.py`.
+
+1-  🗃 Create a new file inside `src/transformers/quantizers/` named `quantizer_your_method.py` and make it inherit from `src/transformers/quantizers/base.py::HFQuantizer`. Make sure to add the new quantizer and quantization config in the quantization auto-mapping in `src/transformers/quantizers/auto.py`
+
+2- 🔩 Define the following class attributes / property methods:
+
+2.1. `requires_calibration`: Whether the quantization method requires a data-calibration process. If set to `True` you'll be able to only support inference (with quantized weights) and not inference + quantization.
```

Comment:
Let's make these bullet points since it doesn't necessarily have to be sequential

```suggestion
* `requires_calibration`: Whether the quantization method requires a data calibration process. If set to `True`, you can only support inference (with quantized weights) and not inference and quantization.
```